### ISSUE TITLE
Better CI for docs

### ISSUE THREAD

**Issue by ivirshup** [scverse/scanpy#1144]
@flying-sheep @falexwolf 
 
[Doc builds are broken](https://readthedocs.com/projects/icb-scanpy/builds/310089/) ... again. This really has to be part of the CI process. It should be implemented in a way where the CI marks a PR as failing if read the docs would error.

I see two options:

* Get the readthedocs CI builds (which we can ask them about)
* Figure out how to make our current travis builds error like read the docs does.

@giovp I don't believe you can mark keys within a dictionary as attributes. I think that's causing this error:

```python
Warning, treated as error:
/home/docs/checkouts/readthedocs.org/user_builds/icb-scanpy/checkouts/latest/scanpy/readwrite.py:docstring of scanpy.read_visium:38:py:attr reference target not found: anndata.AnnData.uns['spatial']['library_id']
```

**Comment by giovp** [comment]
Sorry for that @ivirshup ! I saw you already fixed it in #1145 , thank you and I will keep it in mind for next time.

**Comment by ivirshup** [comment]
We've got CI doc builds now!
